### PR TITLE
Close Etcd client when process exits

### PR DESCRIPTION
**Pull request by jiacheliu3**
### What changes are proposed in this pull request?

1. Since EtcdClient is a singleton, close it in the jvm shutdownHook
2. Close related resources within EtcdClient

### CODE REVIEW

File: dora/core/common/src/main/java/alluxio/membership/ServiceDiscoveryRecipe.java [hunk 1]
```diff
@@ -70,7 +70,7 @@ public ServiceDiscoveryRecipe(AlluxioEtcdClient client, String pathPrefix) {
     mAlluxioEtcdClient = client;
     mRegisterPathPrefix = pathPrefix;
     mExecutor = Executors.newSingleThreadScheduledExecutor(
-        ThreadFactoryUtils.build("service-discovery-checker", false));
+        ThreadFactoryUtils.build("service-discovery-checker", true));
```
Comment (jiacheliu3) [inline dora/core/common/src/main/java/alluxio/membership/ServiceDiscoveryRecipe.java:73]: use daemon thread so the existence of the thread does not block process exit
https://stackoverflow.com/questions/2213340/what-is-a-daemon-thread-in-java

**Comment by Xenorith** [comment]
```
2023-09-25T03:43:05.8726490Z [ERROR] /usr/src/alluxio/dora/core/common/src/main/java/alluxio/membership/AlluxioEtcdClient.java:550:3: Missing a Javadoc comment. [MissingJavadocMethod]
2023-09-25T03:43:05.8727749Z [ERROR] /usr/src/alluxio/dora/core/common/src/main/java/alluxio/membership/ServiceDiscoveryRecipe.java:354:3: Missing a Javadoc comment. [MissingJavadocMethod]
```

**Review by huanghua78 — APPROVED** [review]
Looks good.

**Review by dbw9580 — APPROVED** [review]
LGTM

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**